StockTrader/src/main/java/com/restResource/StockTrader/repository/BuyTriggerRepository.java

```diff
@@ -20,7 +20,16 @@
     @Query(value =
             "UPDATE buy_trigger " +
             "SET stock_amount = buy_trigger.stock_amount + ?2 " +
-            "WHERE user_id = ?1 ",
+            "WHERE buy_trigger.user_id = ?1 AND buy_trigger.stock_symbol = ?3 ",
             nativeQuery = true)
-    Integer incrementStockAmount(String userId, Integer amount);
+    Integer incrementStockAmount(String userId, Integer amount, String stockSymbol);
+
+    @Transactional
+    @Modifying
+    @Query(value =
+            "UPDATE buy_trigger " +
+             "SET stock_cost = ?2 " +
+             "WHERE buy_trigger.user_id = ?1 AND buy_trigger.stock_symbol = ?3 AND buy_trigger.cost = NULL ",
```

Comment:
should be cost IS NULL. see [here](https://www.postgresql.org/docs/8.1/functions-comparison.html)